packages/instant/src/components/zero_ex_instant.tsx

```diff
@@ -52,15 +62,24 @@ export class ZeroExInstant extends React.Component<ZeroExInstantProps> {
             assetBuyer,
             network,
             selectedAsset: assetUtils.createAssetFromAssetData(props.assetData, completeAssetMetaDataMap, network),
+            selectedAssetAmount: _.isUndefined(props.defaultAssetBuyAmount)
+                ? state.selectedAssetAmount
+                : new BigNumber(props.defaultAssetBuyAmount),
             assetMetaDataMap: completeAssetMetaDataMap,
         };
         return storeStateFromProps;
     }
     constructor(props: ZeroExInstantProps) {
         super(props);
-        this._store = store.create(ZeroExInstant._mergeInitialStateWithProps(this.props, INITIAL_STATE));
+        const initialAppState = ZeroExInstant._mergeInitialStateWithProps(this.props, INITIAL_STATE);
+        this._store = store.create(initialAppState);
+    }
+
+    public componentDidMount(): void {
```

Comment:
do we want to do this for will / did change props lifecycle method as well?

Reply:
I think our current position is that we do not support these settings changing after an initial render (had this discussion in https://github.com/0xProject/0x-monorepo/pull/1159). However, I do see how it could be useful. Would rather it be done in a follow-up branch however.